### ISSUE TITLE
Why can't I use yolov3-tiny.weights to train custom objects ?

### ISSUE THREAD

**Issue by wangshenqiang123456** [pjreddie/darknet#1846]
In my project, I modified yolov3-tiny.cfg and used yolov3-tiny.weigths as my pre-training file .

When I use the head of human in the picture as a learning object , the training was normal . And I got a very well final weights. 
But when I change the train object to something else (head to other), I can't train it .

At first I suspected that the sample data maybe wrong , because I used the same marking method and the same cfg file (only voc.data and voc.names are modified accordingly). But when I change the content of the name node in these train samples to head , there were no other configuration or parameter changes ( just the node in xml ,of course voc.data and voc.names has been modified accordingly ) ,  I can train it .
** Even if I change head to head1, I won't be able to train ! **

I don't know why , and I need your help . Below is partial log , do you know what went wrong ?

yolov3-tiny
layer filters size input output
0 conv 16 3 x 3 / 1 480 x 480 x 3 -> 480 x 480 x 16 0.199 BFLOPs
1 max 2 x 2 / 2 480 x 480 x 16 -> 240 x 240 x 16
2 conv 32 3 x 3 / 1 240 x 240 x 16 -> 240 x 240 x 32 0.531 BFLOPs
3 max 2 x 2 / 2 240 x 240 x 32 -> 120 x 120 x 32
4 conv 64 3 x 3 / 1 120 x 120 x 32 -> 120 x 120 x 64 0.531 BFLOPs
5 max 2 x 2 / 2 120 x 120 x 64 -> 60 x 60 x 64
6 conv 128 3 x 3 / 1 60 x 60 x 64 -> 60 x 60 x 128 0.531 BFLOPs
7 max 2 x 2 / 2 60 x 60 x 128 -> 30 x 30 x 128
8 conv 256 3 x 3 / 1 30 x 30 x 128 -> 30 x 30 x 256 0.531 BFLOPs
9 max 2 x 2 / 2 30 x 30 x 256 -> 15 x 15 x 256
10 conv 512 3 x 3 / 1 15 x 15 x 256 -> 15 x 15 x 512 0.531 BFLOPs
11 max 2 x 2 / 1 15 x 15 x 512 -> 15 x 15 x 512
12 conv 1024 3 x 3 / 1 15 x 15 x 512 -> 15 x 15 x1024 2.123 BFLOPs
13 conv 256 1 x 1 / 1 15 x 15 x1024 -> 15 x 15 x 256 0.118 BFLOPs
14 conv 512 3 x 3 / 1 15 x 15 x 256 -> 15 x 15 x 512 0.531 BFLOPs
15 conv 21 1 x 1 / 1 15 x 15 x 512 -> 15 x 15 x 21 0.005 BFLOPs
16 yolo
17 route 13
18 conv 128 1 x 1 / 1 15 x 15 x 256 -> 15 x 15 x 128 0.015 BFLOPs
19 upsample 2x 15 x 15 x 128 -> 30 x 30 x 128
20 route 19 8
21 conv 256 3 x 3 / 1 30 x 30 x 384 -> 30 x 30 x 256 1.593 BFLOPs
22 conv 18 1 x 1 / 1 30 x 30 x 256 -> 30 x 30 x 18 0.008 BFLOPs
23 yolo
Loading weights from yolov3-tiny.weights...Done!
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Resizing
320
Loaded: 0.000028 seconds
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.006367, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502280, .5R: -nan, .75R: -nan, count: 0
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.005875, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502278, .5R: -nan, .75R: -nan, count: 0
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.006479, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502270, .5R: -nan, .75R: -nan, count: 0
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.005764, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502274, .5R: -nan, .75R: -nan, count: 0
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.005963, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502267, .5R: -nan, .75R: -nan, count: 0
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.006455, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502243, .5R: -nan, .75R: -nan, count: 0
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.006555, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502268, .5R: -nan, .75R: -nan, count: 0
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.006006, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.502296, .5R: -nan, .75R: -nan, count: 0
500209: 151.435257, 151.435257 avg, 0.001000 rate, 0.310133 seconds, 32013376 images
Loaded: 0.000042 seconds

**Comment by alanMachineLeraning** [comment]
@wangshenqiang123456  I wonder how you solved this problem?

**Comment by wangshenqiang123456** [comment]
@alanMachineLeraning 看你名字应该能看懂中文，就不用英语答复你了。
我这个问题是因为负责这块的同事离职了，接手时他也没有过多的向我介绍他所做的改动， 我一开始以为只是voc.data和voc.names这两个文件会指定训练对象，但后来发现他在制作样本txt文件的python中也对训练对象做了指定，所以开始时不知道问题在哪比较困惑，后来我详细的看了一遍整个样本的制作过程，就找到问题了

**Comment by wangshenqiang123456** [comment]
@alanMachineLeraning 看你名字应该能看懂中文，就不用英语答复你了。
我这个问题是因为负责这块的同事离职了，接手时他也没有过多的向我介绍他所做的改动， 我一开始以为只有voc.data和voc.names这两个文件会指定训练对象，所以比较困惑问题出在哪，后来我详细的看了一遍整个样本的制作过程，发现他在制作样本txt文件的python中也对训练对象做了指定，就找到问题了

**Comment by bingnoi** [comment]
博主您好，请问在您的案例中有哪些地方需要调整呢？

**Comment by wangshenqiang123456** [comment]
学生党吧，大年初一的，还这么用功？
我了解的也只是浅浅的应用，我现在在老家没有网络，只能靠记忆用手机邮件回复你。
网上有很多关于这一块设置的博客；下面说说我的记忆吧，时间也挺久的，记得不清楚，希望能对你有帮助。
准备学习样本（jpg和txt）和修改样本相关配置文件(记不太清，如voc.data,voc.names这些)，如果是Linux下的还有c文件指定的路径要修改，等等。我就当你都OK了，使用yolov3-tiny.weights需要修改对应cfg的几个常见地方博客也很容易搜到。
回归正题，忘了这是针对哪一个平台的了。
我当初在Linux下的这些配置都是我这位置上的前职员弄好了的，他离职了我再入职的，没有交接，导致我没注意到他制作样本时，在相关的Python文件中把检测对象写死了，我发现之后改完就可以训练了。
由于一开始我并没有察觉到问题在这，所以考虑了换平台，于是在Windows上也做了配置，只要显卡够用，这个配置几乎完全和官网一致就没有什么问题了（Windows平台官网配置方法确实相对详细或简单一点）
最后，如果你觉得所有的你都弄OK了，还是没法训练，那可能就是你的GPU不够用了，如果真的是显卡问题把.cfg文件中的random＝0试试。

---Original---
